distributed/comm/utils.py

```diff
@@ -18,6 +18,16 @@
     OFFLOAD_THRESHOLD = parse_bytes(OFFLOAD_THRESHOLD)
 
 
+# Find the function, `host_array()`, to use when allocating new host arrays
+try:
+    # Use NumPy, when available, to avoid memory initialization cost
+    import numpy
```

Comment:
We'd been hoping to avoid importing NumPy when it's not needed (https://github.com/dask/distributed/issues/5729). This change feels like a fine reason to me to say "NumPy is a required import of distributed" and give up on that goal, but wanted to note it. I suppose we could defer the import into the `host_array` function, but that doesn't really gain us anything. cc @crusaderky

Reply:
Deferring the `import` to a function would just ensure that line is run every time we create a buffer, which adds a (small) performance hit (though larger on the first `read`).

Reply:
Exactly. I think we should leave the import at the top-level, just wanted to point it out.